openedx/core/djangoapps/catalog/management/commands/sync_course_runs.py

```diff
@@ -0,0 +1,55 @@
+"""
+Sync course runs from catalog service.
+"""
+import logging
+
+from django.core.management.base import BaseCommand
+from opaque_keys.edx.keys import CourseKey
+
+from catalog.utils import get_catalog_course_runs
+from content.course_overviews.models import CourseOverview
+
+logger = logging.getLogger(__name__)
+
+
+def fetch_catalog_course_runs():
+    """
+    Fetch catalog course runs from catalog service for all course overview records.
+    """
+    course_overviews = CourseOverview.objects.all()
```

Comment:
Here are 2 others reason to consider (now that number 1-3 have been invalidated):
4. **Paging.** Given that `CourseOverview` is simply an LMS cache of the course-metadata, and the `CourseCatalog` is the source of truth, I would expect the indexing/paging from the Catalog API to be a better one to use if/when we choose to parallelize/break this management command into multiple subtasks.

5. **Performance/Scalability.** Rather than fetching all the data/rows from the `CourseOverviews` table and then making an in-memory list of all the courses to eventually pass to `CourseCatalog`, which then needs to filter by all those courses, I expect the system would behave better to simply have the `CourseCatalog` paginate all its data.